backends/p4tools/p4tools.def

```diff
@@ -144,7 +144,12 @@ class Extracted_Varbits : Type_Bits {
  public:
 #emit
     void *operator new(size_t size) { return ::operator new(size); }
+// FIXME: Remove this #ifdefine check once we switch to C++20
+#if defined(__cpp_sized_deallocation) && __cpp_sized_deallocation >= 201309L
     void operator delete(void *p, size_t size) { return ::operator delete(p, size); }
+#else
+    void operator delete(void *p) { return ::operator delete(p); }
+#endif
```

Comment:
Yeah. Does not seem related to me.

Reply:
Based on the discussion in https://reviews.llvm.org/D112921 I gather that recent versions of clang will support size deallocation. Once we switch to C++20 and upgrade our compilers it will be worth revisiting whether we still need this. The C++20 is just an easily greppable tag.

Reply:
The problem being compilers that claim to support a certain version of C++ (14 or 20) but don't completely -- they're missing some features.  The `__cpp_FETAURE` macros themselves are part of C++23, so there exists the possibility of a C++ compiler that supports the feature but does not define the macro, but hopefully that will be rare.

Reply:
Note that we could just use `#if __cpp_sized_deallocation >= 201309L` as the preprocessor defines any identifier that is not a macro evaluates to 0 in an `#if`

Reply:
I decided to leave it as `defined(__cpp_sized_deallocation) && __cpp_sized_deallocation >= 201309L` to make this check a bit more explicit.

Reply:
> Based on the discussion in https://reviews.llvm.org/D112921 I gather that recent versions of clang will support size deallocation. Once we switch to C++20 and upgrade our compilers it will be worth revisiting whether we still need this. The C++20 is just an easily greppable tag.

This is enabled by default in mainline clang as of LLVM 19. However, sized deallocation is a platform choice. E.g. on Apple downstream clang fork it will definitely be availability-tied and enabled only in certain SDK versions.